File: SQLMemTable/Demos/C++ Builder 6/ExportToSQL/Unit1.cpp
//---------------------------------------------------------------------------
#include <vcl.h>
#pragma hdrstop

#include "Unit1.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "SQLMemMain"
#pragma link "SQLMemMain"
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button2Click(TObject *Sender)
{
 Close();
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button3Click(TObject *Sender)
{
 SQLMemQuery1->SQL->Text = Memo1->Text;
 SQLMemQuery1->ExecSQL();
 ShowMessage("Script executed successfully, rows affected = "+IntToStr(SQLMemQuery1->RowsAffected));
}
//---------------------------------------------------------------------------
void __fastcall TForm1::Button1Click(TObject *Sender)
{
 AnsiString s;
 TFileStream* fs;
 int i;

 s = "";
 if (lbTables->SelCount > 0)
  for (i = 0; i < lbTables->Items->Count; i++)
   if (lbTables->Selected[i])
    {
     SQLMemTable1->TableName = lbTables->Items->Strings[i];
     s = s + SQLMemTable1->ExportTableToSQL(
      cbExportStructure->Checked,
      cbAddDROPTable->Checked,
      cbExportIndexes->Checked,
      cbAddDROPIndex->Checked,
      cbExportData->Checked,
      cbExportBLOBFields->Checked,
      cbUseBrackets->Checked);
    };
 Memo1->Text = s;
 fs = new TFileStream("test.sql",fmCreate);
 fs->WriteBuffer(&s[1],s.Length());
 delete fs;
}
//---------------------------------------------------------------------------
void __fastcall TForm1::FormCreate(TObject *Sender)
{
 SQLMemQuery1->ExecSQL();
 SQLMemTable1->GetTableNames(lbTables->Items);
}
//---------------------------------------------------------------------------

File: SQLMemTable/Demos/Delphi/ExportToSQL/Unit1.pas
unit Unit1;

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms,
  Dialogs, DB, StdCtrls, ComCtrls, ExtCtrls, SQLMemMain;

type
  TForm1 = class(TForm)
    Panel1: TPanel;
    Panel2: TPanel;
    Button1: TButton;
    Button3: TButton;
    Button2: TButton;
    Panel4: TPanel;
    Label1: TLabel;
    gbSQL: TGroupBox;
    Memo1: TRichEdit;
    PageControl1: TPageControl;
    tsTables: TTabSheet;
    tsExportOptions: TTabSheet;
    gbTables: TGroupBox;
    lbTables: TListBox;
    gbExportOptions: TGroupBox;
    GroupBox3: TGroupBox;
    cbExportStructure: TCheckBox;
    cbAddDROPTable: TCheckBox;
    GroupBox4: TGroupBox;
    cbExportData: TCheckBox;
    cbExportBLOBFields: TCheckBox;
    GroupBox5: TGroupBox;
    cbExportIndexes: TCheckBox;
    cbAddDROPIndex: TCheckBox;
    cbUseBrackets: TCheckBox;
    Splitter1: TSplitter;
    SQLMemTable1: TSQLMemTable;
    SQLMemQuery1: TSQLMemQuery;
    procedure Button2Click(Sender: TObject);
    procedure FormCreate(Sender: TObject);
    procedure Button1Click(Sender: TObject);
    procedure Button3Click(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.dfm}

procedure TForm1.Button2Click(Sender: TObject);
begin
 Close;
end;

procedure TForm1.FormCreate(Sender: TObject);
begin
 SQLMemQuery1.ExecSQL;
 SQLMemTable1.GetTableNames(lbTables.Items);
end;

procedure TForm1.Button1Click(Sender: TObject);
var s:  String;
    fs: TFileStream;
    i:  Integer;
begin
 s := '';
 if (lbTables.SelCount > 0) then
  for i := 0 to lbTables.Count - 1 do
   if (lbTables.Selected[i]) then
    begin
     SQLMemTable1.TableName := lbTables.Items[i];
     s := s + SQLMemTable1.ExportTableToSQL(
      cbExportStructure.Checked,
      cbAddDROPTable.Checked,
      cbExportIndexes.Checked,
      cbAddDROPIndex.Checked,
      cbExportData.Checked,
      cbExportBLOBFields.Checked,
      cbUseBrackets.Checked);
    end;
 Memo1.Text := s;
 fs := TFileStream.Create('test.sql',fmCreate);
 fs.WriteBuffer(PChar(@s[1])^,Length(s));
 fs.Free;
end;

procedure TForm1.Button3Click(Sender: TObject);
begin
 SQLMemQuery1.SQL.Text := Memo1.Text;
 SQLMemQuery1.ExecSQL;
 ShowMessage('Script executed successfully, rows affected = '+IntToStr(SQLMemQuery1.RowsAffected));
end;

end.
